homeassistant/components/elmax/sensor.py

```diff
@@ -0,0 +1,71 @@
+"""Elmax sensor platform."""
+from __future__ import annotations
+
+from elmax_api.model.panel import PanelStatus
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ElmaxCoordinator
+from .common import ElmaxEntity
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Elmax sensor platform."""
+    coordinator: ElmaxCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    known_devices = set()
+
+    def _discover_new_devices():
+        panel_status: PanelStatus = coordinator.data
+        # In case the panel is offline, its status will be None. In that case, simply do nothing
+        if panel_status is None:
+            return
```

Comment:
Maybe you should throw PlatformNotReady here? However it would make more sense to move this to init perhaps and throw ConfigEntryNotReady.

In which situation can this be None and how can the integration recover from this?

Reply:
This is None when the Elmax cloud reports the panel as OFFLINE. Having a panel in offline state does not necessarily mean that the integration is not ready. The current implementation of the coordinator holds a NONE value (as data) in case the cloud reports the panel is offline. Notice: the integration is ready (the cloud is responding correctly), but the panel is offline, so we cannot retrieve updated state of the connected entities to that panel. 

As soon as the panel gets online, the Coordinator will hold a not-none object (panel_status), and this if-statement won't be entered. 

I hope this clarifies things a bit for you.

Reply:
If the panel is offline, shouldn't you mark all relevant entities as unavailable? `_attr_available`?

Reply:
And that's exactly what already happens [here](https://github.com/albertogeniola/core/blob/b248073ff4e4f1f5ac3e83eaa5dc096a21de1543/homeassistant/components/elmax/common.py#L172-L175) :)

```python
@property
def available(self) -> bool:
    """Return if entity is available."""
    return super().available and self._panel.online
```

Please note, once again, that the `_discover_new_device()` method is called by the coordinator at every update. The method is invoked regardless of the the online status of the panel (as it's issued against the Elmax cloud) but the panel-data is only available when the given panel is online. That is why we need to check for null values here.